src/d3d12.zig

```diff
@@ -471,10 +471,29 @@ pub const Device = struct {
         return ShaderModule.initAir(device, air);
     }
 
-    pub fn createShaderModuleSpirv(device: *Device, code: []const u8) !*ShaderModule {
+    pub fn createShaderModuleSpirv(device: *Device, code: [*]const u32, code_size: u32) !*ShaderModule {
         _ = code;
+        _ = code_size;
         _ = device;
-        return error.unsupported;
+        return error.Unsupported;
+    }
+
+    pub fn createShaderModuleHLSL(device: *Device, code: []const u8) !*ShaderModule {
+        _ = device;
+        const module = try allocator.create(ShaderModule);
+        module.* = .{ .code = .{ .code = code } };
+        return module;
+    }
+
+    pub fn createShaderModuleMSL(
+        device: *Device,
+        code: []const u8,
+        workgroup_size: sysgpu.ShaderModule.WorkgroupSize,
```

Comment:
Maybe should be optional as it's only needed for compute?

Reply:
good catch!

Reply:
but anyway it's a little awkward, i think it's time to start refactoring the API